Release 2.2.0

ialokim opened the pull request:
Here are the steps to follow when preparing a new release (some custom tasks added for this release). Please check the following boxes with an `x` when done:

* [x] update PTE to the latest commit at [opentransitmap](https://gitlab.com/opentransitmap/public-transport-enabler/-/commits/master) and run `./update-dependency-pinning.sh`
* [x] re-enable SBB (Switzerland) with new data source #768 
* [x] ~~re-enable PL (Poland) with new data source https://gitlab.com/opentransitmap/public-transport-enabler/-/commit/693d84e340f1525366e6b29a32c0a81401f326ff~~ (needs more investigation as to which area is actually covered)
* [x] test the app (automatically using AndroidTest as well as manually), paying special attention to changes since the last release
* [x] fix any bugs if necessary
* [x] update translations from [Transifex](https://www.transifex.com/) using `tx pull --mode=developer -a` in the root folder (you need proper permissions to do that)
* [x] add newly supported languages to the two arrays in [`app/src/main/res/values/arrays.xml`](https://github.com/grote/Transportr/blob/master/app/src/main/res/values/arrays.xml#L16)
* [x] revise the commits to `master` since the last release and add interesting changes (as well as new languages) to the changelog at [`app/src/main/res/xml/changelog_master.xml`](https://github.com/grote/Transportr/blob/master/app/src/main/res/xml/changelog_master.xml), then run `python3 ./fastlane/generate_changelog.py`
* [x] bump the [`versionCode`](https://github.com/grote/Transportr/blob/master/app/build.gradle#L14) and [`versionName`](https://github.com/grote/Transportr/blob/master/app/build.gradle#L15) in `app/build.gradle`
* [ ] re-enable Transportr for inclusion in F-Droid #826
* [ ] file another PR updating all dependencies which should be merged *after* the successful release

Please also refer to #689 as an example.

ialokim commented:
> * [x] fix any bugs if necessary

several issues reported: #889 #890 https://gitlab.com/opentransitmap/public-transport-enabler/-/issues/6

In order to get the release out soon, I'd consider them as been non-release-blocking.

M00NJ commented:
Could #872 be a part of this release as well?

ialokim commented:
> Could #872 be a part of this release as well?

I don't think this is a good idea, given that we want to publish the release soon, and #872 brings a lot of UI changes that need a thourough testing.

mesinger commented:
@ialokim any way i can help to get this release done?

grote commented on app/src/main/res/xml/changelog_master.xml:
```diff
@@ -2,6 +2,17 @@
+    <release
+        version="2.2.0"
+        versioncode="124">
+        <change>Switch map library to MapLibre - finally bringing Transportr back to F-Droid!</change>
```
Technically, we didn't make it back, yet ;)

ialokim replied:
Yeah it's the next unchecked bullet point in the first post, but we cannot continue without tagging the release :P

ialokim replied:
And we will make it back (I hope)! :smile:

ialokim commented on app/src/androidTest/java/de/grobox/transportr/ScreengrabTest.kt:
```diff
@@ -51,17 +51,13 @@ abstract class ScreengrabTest {
     }
 
-    val networkId: NetworkId = when(Locale.forLanguageTag(getTestLocale())) {
+    val networkId: NetworkId = when(Locale.forLanguageTag(getTestLocale() ?: "de")) {
```
I can't fully remember right now why this was, but I think `getTestLocale()` would return `null` in certain cases and therefore fail the test. Anyway it's just in the test sources which require some overhaul after removing quite some regions, so maybe not a blocker either?

ialokim commented:
Thanks for your review @Altonss!

>     * [ ]  Why updating strings for pl network navitia that is for now removed? IMHO it clutters a bit the PR with useless strings.

There are a lot of leftover strings from now removed regions which should probably be cleaned properly at some point. But you are right, those are unfortunately really useless. I've removed them now manually.

>     * [ ]  Shouldn't we also mention in the changelogs that we've re-added VRS?

Good catch, added that!

grote commented:
Thanks everyone who contributed to this release. You are awesome! :D

I just pressed the button and the release is out. However, Google Play doesn't accept it: `Google Api Error: Invalid request - Target SDK of artifact is too low: 124.` It seems it is currently 32 which won't work anymore. Please see https://support.google.com/googleplay/android-developer/answer/11926878 for details.
